src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_MediaPlayerElement.cs

```diff
@@ -0,0 +1,140 @@
+using System.Windows.Input;
+using FluentAssertions;
+using Microsoft.UI.Xaml.Tests.Common;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Windows.Media.Playback;
+using Windows.Phone.Notification.Management;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Automation.Peers;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using static Private.Infrastructure.TestServices;
+
+namespace Uno.UI.RuntimeTests.Tests.Windows_UI_Xaml_Controls
+{
+	[TestClass]
+	[RunsOnUIThread]
+	public class Given_MediaPlayerElement
+	{
+		[TestMethod]
+		public void When_MediaPlayerElement_Not_SetSource_Property_Value()
+		{
+			var entity = new MediaPlayerElement();
+			entity.SetMediaPlayer(new Windows.Media.Playback.MediaPlayer());
+			entity.MediaPlayer.Play();
+			entity.GetValue(MediaPlayerElement.SourceProperty);
+
+			Assert.IsNull(entity.Source);
+		}
+
+		[TestMethod]
+		public void When_MediaPlayerElement_SetSource_Property_Value()
+		{
+			var entity = new MediaPlayerElement();
+			entity.AutoPlay = true;
+			entity.Source = Windows.Media.Core.MediaSource.CreateFromUri(new Uri("https://test-videos.co.uk/vids/bigbuckbunny/mp4/h264/720/Big_Buck_Bunny_720_10s_5MB.mp4"));
+			entity.GetValue(MediaPlayerElement.SourceProperty);
+			entity.Should().NotBeNull();
+		}
+
+		[TestMethod]
+		public async Task When_MediaPlayerElement_Added_In_Opening()
+		{
+			MediaPlayerElement entity = await StartPlayerAsync();
+
+			entity.MediaPlayer.Play();
+
+			var mediaTransportControls = entity.TransportControls as MediaTransportControls;
+			Assert.IsNotNull(mediaTransportControls);
+
+			var mediaPlayer = entity.MediaPlayer as Windows.Media.Playback.MediaPlayer;
+			Assert.IsNotNull(mediaPlayer);
+		}
+
+		[TestMethod]
+		public void When_MediaPlayerElement_Dependency_Property_Value()
+		{
+			var entity = new MediaPlayerElement();
+			entity.AutoPlay = true;
+			entity.Source = Windows.Media.Core.MediaSource.CreateFromUri(new Uri("https://test-videos.co.uk/vids/bigbuckbunny/mp4/h264/720/Big_Buck_Bunny_720_10s_5MB.mp4"));
+			entity.ApplyTemplate();
+			entity.SetMediaPlayer(new Windows.Media.Playback.MediaPlayer());
+			entity.MediaPlayer.Play();
+
+			entity.GetValue(MediaPlayerElement.SourceProperty);
+			entity.Should().NotBeNull();
+		}
+
+		[TestMethod]
+		public async Task When_MediaPlayerElement_SetIsFullWindow_Check_Fullscreen()
+		{
+
+			MediaPlayerElement entity = await StartPlayerAsync();
+			try
+			{
+
+				entity.IsFullWindow = true;
+
+				Thread.Sleep(3000);
+				entity.GetValue(MediaPlayerElement.IsFullWindowProperty);
+				entity.Should().NotBeNull();
+			}
+			finally
+			{
+				entity.IsFullWindow = false;
+			}
+		}
+
+		[TestMethod]
+		public async Task When_MediaPlayerElement_SetSource_Check_Play()
+		{
+			MediaPlayerElement entity = await StartPlayerAsync();
+
+			entity.MediaPlayer.Play();
+
+			Thread.Sleep(3000);
+			Assert.IsTrue(entity.MediaPlayer.PlaybackSession.IsPlaying);
```

Comment:
no need to wait the full 3s, we can exit as soon as the condition is satifisfied:
```suggestion
			await WindowHelper.WaitFor(() => entity.MediaPlayer.PlaybackSession.IsPlaying, 3000, "Timeout waiting for the playback session state changing to playing.")
```

Reply:
updated.